diff -r c1ae5d937de8 src/gnu/neuron_gnu_builtin.h
--- a/src/gnu/neuron_gnu_builtin.h	Sun May 15 12:32:34 2011 -0400
+++ b/src/gnu/neuron_gnu_builtin.h	Mon May 30 02:13:27 2011 +0100
@@ -93,7 +93,7 @@
 #endif
 #if !defined(IV)
 
-#ifndef __xlC__
+#if !defined (__xlC__) && !defined(ANDROID)
 #ifndef __OBJECTCENTER__
 #ifndef __hpux /* hpux defines this in math.h */
 #ifndef ultrix /* ultrix defines this in math.h */
diff -r c1ae5d937de8 src/ivos/ivstrm.h
--- a/src/ivos/ivstrm.h	Sun May 15 12:32:34 2011 -0400
+++ b/src/ivos/ivstrm.h	Mon May 30 02:13:27 2011 +0100
@@ -40,9 +40,11 @@
 #if defined(HAVE_STREAM_H)
 #include <stream.h>
 #else
+#ifndef ANDROID
 #define _STREAM_COMPAT
 #include <iostream.h>
 #endif
+#endif
 
 // for some compilers stream.h is insufficient
 // following for gcc-3.0.1
diff -r c1ae5d937de8 src/ivos/regexp.cpp
--- a/src/ivos/regexp.cpp	Sun May 15 12:32:34 2011 -0400
+++ b/src/ivos/regexp.cpp	Mon May 30 02:13:27 2011 +0100
@@ -1210,6 +1210,8 @@
 
 static void
 regerror(char* s) {
-	cerr << "regexp: " << s << "\n";
+    #ifndef ANDROID
+    cerr << "regexp: " << s << "\n";
+    #endif
 }
 
diff -r c1ae5d937de8 src/modlunit/units.c
--- a/src/modlunit/units.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/modlunit/units.c	Mon May 30 02:13:27 2011 +0100
@@ -13,6 +13,9 @@
 #if defined(WIN32)
 #include <windows.h>
 #endif
+#ifdef ANDROID
+void signal(int errc, void* errf) {}
+#endif
 
 int	unitonflag = 1;
 static int	UnitsOn = 0;
diff -r c1ae5d937de8 src/nmodl/solve.c
--- a/src/nmodl/solve.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/nmodl/solve.c	Mon May 30 02:13:27 2011 +0100
@@ -35,6 +35,13 @@
 static List *solvq;	/* list of the solve statement locations */
 int numlist = 0;	/* number of slist's */
 
+#ifdef ANDROID
+double atof(const char *str) {
+    char* endptr;
+    return strtod(str, endptr);
+}
+#endif
+
 solvequeue(q1, q2, blocktype, qerr) /*solve NAME=q1 [using METHOD=q2]*/
 				/* q2 = 0 means method wasn't there */
 				/* qerr in ITEM0 or else the closing
diff -r c1ae5d937de8 src/nrnoc/eion.c
--- a/src/nrnoc/eion.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/nrnoc/eion.c	Mon May 30 02:13:27 2011 +0100
@@ -2,6 +2,9 @@
 /* /local/src/master/nrn/src/nrnoc/eion.c,v 1.10 1998/02/26 16:42:50 hines Exp */
 
 #include 	<stdlib.h>
+#ifdef ANDROID
+#include <stdio.h>
+#endif
 #include	"section.h"
 #include	"neuron.h"
 #include	"membfunc.h"
diff -r c1ae5d937de8 src/nrnoc/fadvance.c
--- a/src/nrnoc/fadvance.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/nrnoc/fadvance.c	Mon May 30 02:13:27 2011 +0100
@@ -3,6 +3,9 @@
 #include <nrnmpi.h>
 #include <nrnrt.h>
 #include <errno.h>
+#ifdef ANDROID
+#include <stdio.h>
+#endif
 #include "neuron.h"
 #include "section.h"
 #include "membfunc.h"
diff -r c1ae5d937de8 src/oc/parse.y
--- a/src/oc/parse.y	Sun May 15 12:32:34 2011 -0400
+++ b/src/oc/parse.y	Mon May 30 02:13:27 2011 +0100
@@ -9,6 +9,10 @@
 #pragma alloca
 #endif
 
+#ifdef ANDROID
+#include <stdio.h>
+#endif
+         
 #include "hoc.h"
 #include "code.h"
 #include "equation.h"
diff -r c1ae5d937de8 src/readline/history.c
--- a/src/readline/history.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/readline/history.c	Mon May 30 02:13:27 2011 +0100
@@ -30,6 +30,11 @@
 #define index strchr
 #endif
 
+#ifdef ANDROID
+#define rindex strrchr
+#define index strchr
+#endif
+
 #if __STDC__
 #include <string.h>
 #if HAVE_MALLOC_H
diff -r c1ae5d937de8 src/readline/readline.c
--- a/src/readline/readline.c	Sun May 15 12:32:34 2011 -0400
+++ b/src/readline/readline.c	Mon May 30 02:13:27 2011 +0100
@@ -36,6 +36,11 @@
 #define index strchr
 #endif
 
+#ifdef ANDROID
+#define rindex strrchr
+#define index strchr
+#endif
+
 #if __STDC__
 #if HAVE_MALLOC_H
 #include <malloc.h>
@@ -126,7 +131,11 @@
 #endif
 
 #include <errno.h>
+#ifndef ANDROID
 extern int errno;
+#else
+extern volatile int errno;
+#endif
 
 #include <setjmp.h>
 #include <sys/stat.h>
@@ -3910,15 +3919,18 @@
 
       username = savestring (&text[first_char_loc]);
       namelen = strlen (username);
+#ifndef ANDROID      
       setpwent ();
+#endif
     }
 
+#ifndef ANDROID      
   while (entry = getpwent ())
     {
       if (strncmp (username, entry->pw_name, namelen) == 0)
 	break;
     }
-
+#endif
   if (!entry)
     {
       endpwent ();
